app/database.js

```diff
@@ -70,27 +72,41 @@ define(['sqlite', 'app/config'], function (sqlite, config) {
 		},
 		executeQuery : function (queryname, args) {
 			var query = queries[queryname] ;
-			db.run(query, args, onError);
+			db.run(query, args);
 		},
-		selectDatasetItems :  function (queryname, args) {
-			var query = queries[ queryname] ;
+		getQuery: function(queryname,args){
+			var queryResult = [];
+			var query = queries[queryname] ;
 			db.each(query,args, function(row, err) {
-				console.log(row.item_answer );
+				queryResult.push(row);
 			});
+			return queryResult;
 		},
-		selectUserSubjects :  function (queryname, args) {
-			var query = queries[ queryname] ;
+		getUnique: function(queryname,unique_name, args) {
+			var queryResult = [];
+			var query = queries[queryname] ;
 			db.each(query,args, function(row, err) {
-				console.log(row.subject_name );
+				unique = 1;
+				for (i = 0; i<queryResult.length;i++) {
```

Comment:
Encapsulate this loop in a function, I had to think too long about what you were doing here.